### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by sirh3e**
As discussed in #85

### CODE REVIEW

**Comment by Gsantomaggio** [comment]
You need to execute `dotnet format` this is why the CI fails.
I need to add the documentation/make files for that. 
Thank you

**Comment by Gsantomaggio** [comment]
@sirh3e Can you please give me the grant to push. I need to change the `approval.txt` file
thank you.

p.s. I will squash all the commits in one!

**Comment by sirh3e** [comment]
@Gsantomaggio Sorry, but I'm not sure whether I should grant you permission on this pr or on my fork?

**Comment by lukebakken** [comment]
@sirh3e please follow these instructions to allow pushes to your fork for your PR branches:

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

That way we can clean up for you. Thanks!

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,10 @@
 all: test build
 
 
-build: 
+format:
+	dotnet format
+
+build: format
```
Comment (lukebakken) [inline Makefile:7]: I'd rather not have the `build` target depend on the `format` target.

**Review by lukebakken — CHANGES_REQUESTED** [review]

**Comment by sirh3e** [comment]
image

As you see in this image there is no checkbox to check for "Allow edits from maintainers"

Doc Ref:
image

**Comment by lukebakken** [comment]
That's strange! OK I will open a new PR with your changes, one minute.

**Comment by lukebakken** [comment]
Please see #91

**Pull request closed without merging**